Enigmatry.Entry.AspNetCore.Authorization/Requirements/UserHasPermission.cs

```diff
@@ -0,0 +1,73 @@
+﻿using Enigmatry.Entry.AspNetCore.Authorization.Attributes;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.Logging;
+
+namespace Enigmatry.Entry.AspNetCore.Authorization.Requirements;
+
+#pragma warning disable CA1711 // Identifiers should not have incorrect suffix
+public static class UserHasPermission
+#pragma warning restore CA1711 // Identifiers should not have incorrect suffix
+{
+    public class Requirement : IAuthorizationRequirement { }
+
+    public class RequirementHandler : AuthenticatedUserRequirementHandler<Requirement>
+    {
+        private readonly IAuthorizationProvider _authorizationProvider;
+
+        public RequirementHandler(
+            IAuthorizationProvider authorizationProvider,
+            ILogger<AuthenticatedUserRequirementHandler<Requirement>> logger)
+            : base(logger)
+        {
+            _authorizationProvider = authorizationProvider;
+        }
+
+        protected override async Task HandleRequirementAsync(AuthorizationHandlerContext context, Requirement requirement)
+        {
+            await base.HandleRequirementAsync(context, requirement);
+
+            if (context.HasFailed)
+            {
+                return;
```

Comment:
HandleRequirementAsync is identical in both classes. The only difference is i.e. UserHasPermission invocation. Logic can be moved to base class and this derivation can be renamed to i.e. FulfilsRequirements() and be implemented differently in derived classes.